.pre-commit-config.yaml

```diff
@@ -0,0 +1,64 @@
+# This is the configuration file for pre-commit (https://pre-commit.com/).
+# To use:
+# * Install pre-commit (https://pre-commit.com/#installation)
+# * Copy this file as ".pre-commit-config.yaml"
+# * Run "pre-commit install".
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.1.0
+    hooks:
+      - id: check-toml
+      - id: check-yaml
+      - id: debug-statements
+      - id: end-of-file-fixer
+      - id: mixed-line-ending
+        args: [ "--fix=lf" ]
+      - id: trailing-whitespace
+
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v2.30.1
+    hooks:
+      - id: pyupgrade
+        args: [ "--py36-plus", "--keep-mock" ]
+
+  - repo: https://github.com/pre-commit/mirrors-autopep8
+    rev: v1.6.0
+    hooks:
+      - id: autopep8
+
+  - repo: https://github.com/pycqa/isort
+    rev: 5.10.1
+    hooks:
+      - id: isort
+
+  - repo: https://github.com/csachs/pyproject-flake8
+    rev: v0.0.1a2.post1
+    hooks:
+      - id: pyproject-flake8
+
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v0.931
+    hooks:
+      - id: mypy
+        additional_dependencies: [ "pytest", "trio-typing", "packaging" ]
+
+  - repo: local
+    hooks:
+      - id: pyright
+        name: pyright
+        entry: pyright
+        language: node
+        pass_filenames: false
```

Comment:
Why have pyright check all files on every commit but not mypy?

Reply:
I'm confused. Mypy is right above pyright there.

Reply:
I've added pyright back to the GA test workflow and removed it from the pre-commit config.